src/core/server/opensearch/version_check/opensearch_opensearch_dashboards_version_compatability.ts

```diff
@@ -110,6 +110,7 @@ function legacyVersionCompatibleWithOpenSearchDashboards(
     legacyVersionNumbers.major === 7 &&
     legacyVersionNumbers.minor === 10 &&
     legacyVersionNumbers.patch === 2 &&
-    opensearchDashboardsVersionNumbers.major === 1
+    (opensearchDashboardsVersionNumbers.major === 1 ||
+      opensearchDashboardsVersionNumbers.major === 2)
```

Comment:
Do we want to start an object that contains the possible major versions and checking against it rather than chaining  ORs?

Reply:
Sure I can do that. Hopefully we won't have to do this for 3.0 but I can see it taking longer than that.